smdebug/core/utils.py

```diff
@@ -72,6 +81,44 @@ def is_s3(path):
         return False, None, None
 
 
+def is_first_process(path):
+    """
+    This function is used to determine the caller of the process
+    is the first process to do so.
+
+    It uses the os.O_EXCL flag (https://linux.die.net/man/3/open)
+    to fail if the the file already exists, i.e another process has
+    written first.
+    :param path: path to the trial
+    :return: boolean that indicates if the caller was the
+    first process to execute the fn.
+    """
+    ensure_dir(path, is_file=False)
+    filename = os.path.join(path, "claim.smd")
+    s3, _, _ = is_s3(path)
+    if s3:
+        return True  # Cannot Implement This Functionality for S3
+    else:
+        try:
+            fd = os.open(filename, os.O_CREAT | os.O_EXCL | os.O_WRONLY)
+            os.close(fd)
+            return True
+        except FileExistsError:
+            return False
+
+
+def remove_claim_file(path):
+    """
+    This function deletes the claim.smd file created by the is_first_process fn
+    when the hook is closed.
+
+    :param path: path to the trial
+    :return: boolean that indicates if the caller was the
+    first process to execute the fn.
+    """
+    os.remove(os.path.join(path, "claim.smd"))
```

Comment:
if that file exist, then only remove.

Reply:
Documentation is incorrect nothing is returned. Also, suggest to factor out claim.smd as it appears twice as harcoded string.